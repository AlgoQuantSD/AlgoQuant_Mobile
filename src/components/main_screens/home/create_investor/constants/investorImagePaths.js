export const NUM_INVESTOR_IMAGES_PER_FREQ = 3;

export const INVESTOR_IMAGE_BASE_URL =
  "https://algoquant-resources.s3.amazonaws.com/InvestorImages";
