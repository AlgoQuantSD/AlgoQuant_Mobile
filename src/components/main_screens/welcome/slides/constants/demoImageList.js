import DemoInvestorImage0 from "../../../../../../assets/demo_images/investor-0.png";
import DemoInvestorImage1 from "../../../../../../assets/demo_images/investor-1.png";
import DemoInvestorImage2 from "../../../../../../assets/demo_images/investor-2.png";
import DemoInvestorImage3 from "../../../../../../assets/demo_images/investor-3.png";
import DemoInvestorImage4 from "../../../../../../assets/demo_images/investor-4.png";
import DemoInvestorImage5 from "../../../../../../assets/demo_images/investor-5.png";
import DemoInvestorImage6 from "../../../../../../assets/demo_images/investor-6.png";

// List of images to display in the demo carousel
export const DEMO_IMAGES_INVESTOR = [
  {
    image: DemoInvestorImage0,
    caption:
      "Begin by pressing the Create Investor button in the bottom right corner of the screen.",
    id: 0,
  },
  {
    image: DemoInvestorImage1,
    caption: "Name your investor and select an investor type.",
    id: 1,
  },
  {
    image: DemoInvestorImage2,
    caption: "Select indicators and a trade frequency.",
    id: 2,
  },
  {
    image: DemoInvestorImage3,
    caption: "Learn more about indicators by pressing the tooltips.",
    id: 3,
  },
  {
    image: DemoInvestorImage4,
    caption:
      "Profit and loss stop dictates the threshold that must be met in order to cash out.",
    id: 4,
  },
  {
    image: DemoInvestorImage5,
    caption: "Select stock for your investor to consider.",
    id: 5,
  },
  {
    image: DemoInvestorImage6,
    caption:
      "Make sure everything looks good before finalizing the creation process.",
    id: 6,
  },
];
