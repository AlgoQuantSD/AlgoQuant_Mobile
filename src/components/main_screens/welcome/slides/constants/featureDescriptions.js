export const INVESTOR_DESCRIPTION = [
  { text: "Customize your investor to your needs", id: 0 },
  { text: "Use technical analysis or AI to configure your investor", id: 1 },
  { text: "Select what assets you want your investor to track", id: 2 },
  { text: "Create jobs for your investor", id: 3 },
];

export const BACKTEST_DESCRIPTION = [
  { text: "Test your investor with historical market data", id: 0 },
  {
    text: "Compare your investor's performance against typical buy/hold strategies",
    id: 1,
  },
  { text: "View in depth statistics about your investor's performance", id: 2 },
  { text: "Gain confidence in your investor", id: 3 },
];

export const JOB_DESCRIPTION = [
  { text: "Deploy your investor into the market via a job", id: 0 },
  {
    text: "Allocate money to your jobs",
    id: 1,
  },
  { text: "View your job's performance in real time", id: 2 },
];

export const PAPER_TRADING_DESCRIPTION = [
  {
    text: "Connect your Alpaca account via API keys for a realistic paper trading experience",
    id: 0,
  },
  {
    text: "Utilize the AlgoQuant simulated brokerage if you do not wish to connect your Alpaca account",
    id: 1,
  },
  { text: "Monitor your portfolio balance", id: 2 },
];
