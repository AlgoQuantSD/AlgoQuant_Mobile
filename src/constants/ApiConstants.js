import React from "react";

export const TRADE_HISTORY_FETCH_AMOUNT = 10;