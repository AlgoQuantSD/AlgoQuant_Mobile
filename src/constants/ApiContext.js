import { AlgoQuant } from "algoquant/lib/AlgoQuant";
import React from "react";

const algoqant = new AlgoQuant();
const AlgoquantApiContext = React.createContext(algoqant);
export default AlgoquantApiContext;
