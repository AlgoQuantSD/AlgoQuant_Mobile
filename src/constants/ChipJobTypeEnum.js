// Enum for the different chips used for the different job types
export const CHIP_JOB_TYPES = {
  Active: 0,
  Past: 1,
};
