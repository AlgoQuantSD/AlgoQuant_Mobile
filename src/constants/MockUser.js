// The MOCK_USER will serve as our hardcoded user until we can interface with the API
export const MOCK_USER = {
  data: {
    totalBalance: 153012.23,
  },
  alpaca: {
    isConnected: false,
  },
};
