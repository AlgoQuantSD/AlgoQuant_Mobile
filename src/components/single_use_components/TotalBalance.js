import { React, useContext, useState, useEffect } from "react";
import { View, Text, StyleSheet } from "react-native";
import { THEME } from "../../constants/Theme";
import { Ionicons } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native-gesture-handler";
import AlgoquantApiContext from "../../constants/ApiContext";
import LoadSpinner from "../reusable_components/LoadSpinner";
import { formattedBalance } from "../../helpers/formatUserBalance";
import { resetBalanceModalBuilder } from "../../helpers/modalFactory";

export default function TotalBalance(props) {
  function handleResetButtonPress() {
    // The spread operator to add to the prop
    resetBalanceModalBuilder({ ...props, alpacaAccount: alpacaConnection });
  }
  // Format the total balance into a string
  const formattingOptions = {
    style: "currency",
    currency: "USD",
    minimumFractionDigits: 2,
  };
  const dollarString = new Intl.NumberFormat("en-US", formattingOptions);

  const algoquantApi = useContext(AlgoquantApiContext);

  const [balance, setBalance] = useState();
  const [isLoading, setIsLoading] = useState(true);
  const [alpacaConnection, setAlpacaConnection] = useState(false);

  useEffect(() => {
    if (algoquantApi.token) {
      algoquantApi.getUser().then((resp) => {
        setBalance(resp.data.buying_power);
        setAlpacaConnection(resp.data.alpaca);
        setIsLoading(false);
      });
    }
  }, [algoquantApi]);

  return (
    <View style={styles.totalBalanceContainer}>
      <Text style={styles.text}>
        {alpacaConnection
          ? "Alpaca verfied Buying Power"
          : "Simulated Buying Power"}
      </Text>
      <View style={styles.balance}>
        <Text style={styles.balanceText} testID="total-balance">
          {isLoading ? <LoadSpinner /> : dollarString.format(balance)}
        </Text>
        <TouchableOpacity
          hitSlop={{ top: 30, bottom: 30, left: 30, right: 30 }}
          onPress={handleResetButtonPress}
        >
          <Ionicons
            name="refresh"
            size={THEME.text.fontSizeBold}
            color={THEME.colors.foreground}
            testID="reset-icon"
          />
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  totalBalanceContainer: {
    alignItems: "center",
    justifyContent: "center",
    flex: 0.15,
    width: "100%",
  },
  balance: {
    flexDirection: "row",
    paddingTop: "1%",
  },
  balanceText: {
    fontSize: THEME.text.fontSizeBold,
    color: THEME.text.color,
    fontWeight: "bold",
    paddingRight: "1%",
    paddingLeft: "1%",
  },
  text: {
    fontSize: THEME.text.fontSizeBody,
    color: THEME.text.color,
  },
});
