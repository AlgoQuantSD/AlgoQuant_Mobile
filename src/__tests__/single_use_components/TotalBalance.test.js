import React from "react";
import { render } from "@testing-library/react-native";
import TotalBalance from "../../components/single_use_components/TotalBalance";
import { mockUser } from "../../constants/MockUser";

describe("TotalBalance", () => {
  it("Should render", () => {
    const tree = render(<TotalBalance />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders the user's balance", () => {
    const { getByTestId } = render(<TotalBalance />);
    const balanceText = getByTestId("total-balance");
    expect(balanceText).toHaveTextContent(
      new Intl.NumberFormat("en-US", {
        style: "currency",
        currency: "USD",
        minimumFractionDigits: 2,
      }).format(mockUser.data.totalBalance)
    );
  });
  it("renders the reset icon", () => {
    const { getByTestId } = render(<TotalBalance />);
    const resetIcon = getByTestId("reset-icon");
    expect(resetIcon).toBeTruthy();
  });
});
