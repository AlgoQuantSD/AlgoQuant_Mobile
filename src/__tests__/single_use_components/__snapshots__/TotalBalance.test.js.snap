// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.15,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
      }
    }
  >
    Total Balance
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": "1%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingLeft": "1%",
          "paddingRight": "1%",
        }
      }
      testID="total-balance"
    >
      $153,012.23
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      exclusive={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      hitSlop={
        Object {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      touchSoundDisabled={false}
    >
      <View
        accessible={true}
        collapsable={false}
        hitSlop={
          Object {
            "bottom": 30,
            "left": 30,
            "right": 30,
            "top": 30,
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`TotalBalance Should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.15,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
      }
    }
  >
    Total Balance
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingTop": "1%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "bold",
          "paddingLeft": "1%",
          "paddingRight": "1%",
        }
      }
      testID="total-balance"
    >
      $153,012.23
    </Text>
    <RNGestureHandlerButton
      collapsable={false}
      delayLongPress={600}
      enabled={true}
      exclusive={true}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      hitSlop={
        Object {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      touchSoundDisabled={false}
    >
      <View
        accessible={true}
        collapsable={false}
        hitSlop={
          Object {
            "bottom": 30,
            "left": 30,
            "right": 30,
            "top": 30,
          }
        }
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
